openedx/features/wikimedia_features/wikimedia_general/signals.py

```diff
@@ -52,22 +52,50 @@ def create_default_course_mode(sender, instance, created, **kwargs):
         else:
             logger.info('Default mode set is Audit - no need to change course mode.')
 
+@receiver(thread_created)
+def handle_thread_creation_notifications(sender, user, post, **kwargs):
+    """
+    Handles the sending of email notifications upon the creation of a discussion thread.
+
+    This function is triggered when a new discussion thread is created. It fetches the
+    current site context and then delegates the responsibility of sending out different
+    types of email notifications to specific functions.
+
+    Args:
+        sender: The model class that sent the signal.
+        user: The user who created the discussion thread.
+        post: The discussion thread that was created.
+        **kwargs: Additional keyword arguments.
+
+    Note:
+        This function serves as a central point for handling all notifications related
+        to thread creation. It ensures that the current site is determined only once and 
+        then passes this information to the functions responsible for individual 
+        notification types.
+    """
+    current_site = get_current_site()
+    if current_site is None:
+        logger.error("Current site could not be determined.")
+        return
+
+    send_thread_mention_email_notification(sender, user, post, current_site, **kwargs)
+    send_new_post_email_notification_to_instructors(sender, user, post, current_site, **kwargs)
+
 @receiver(comment_created)
 @receiver(comment_edited)
-@receiver(thread_created)
 @receiver(thread_edited)
-def send_thread_mention_email_notification(sender, user, post, **kwargs):
+def send_thread_mention_email_notification(sender, user, post,current_site, **kwargs):
```

Comment:
Please verify comment_created, comment_edited, and thread_edited notifications. We might have to send current_site in these signals.

Reply:
We can add a logic, if current_site in this function is none then get the site from the site model.

Reply:
Feedback addressed